lib/Slack_web_api.js

```diff
@@ -1,314 +1,180 @@
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('groups.list', options, cb);
-            },
-            mark: function(options, cb) {
-                slack_api.callAPI('groups.mark', options, cb);
-            },
-            open: function(options, cb) {
-                slack_api.callAPI('groups.open', options, cb);
-            },
-            rename: function(options, cb) {
-                slack_api.callAPI('groups.rename', options, cb);
-            },
-            setPurpose: function(options, cb) {
-                slack_api.callAPI('groups.setPurpose', options, cb);
-            },
-            setTopic: function(options, cb) {
-                slack_api.callAPI('groups.setTopic', options, cb);
-            },
-            unarchive: function(options, cb) {
-                slack_api.callAPI('groups.unarchive', options, cb);
-            },
-        },
-        im: {
-            close: function(options, cb) {
-                slack_api.callAPI('im.close', options, cb);
-            },
-            history: function(options, cb) {
-                slack_api.callAPI('im.history', options, cb);
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('im.list', options, cb);
-            },
-            mark: function(options, cb) {
-                slack_api.callAPI('im.mark', options, cb);
-            },
-            open: function(options, cb) {
-                slack_api.callAPI('im.open', options, cb);
-            }
-        },
-        mpim: {
-            close: function(options, cb) {
-                slack_api.callAPI('mpim.close', options, cb);
-            },
-            history: function(options, cb) {
-                slack_api.callAPI('mpim.history', options, cb);
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('mpim.list', options, cb);
-            },
-            mark: function(options, cb) {
-                slack_api.callAPI('mpim.mark', options, cb);
-            },
-            open: function(options, cb) {
-                slack_api.callAPI('mpim.open', options, cb);
+
+                if (json.ok) {
+                    return cb(null, json);
+                }
+                return cb(json.error, json);
             }
-        },
-        pins: {
-            add: function(options, cb) {
-                slack_api.callAPI('pins.add', options, cb);
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('pins.list', options, cb);
-            },
-            remove: function(options, cb) {
-                slack_api.callAPI('pins.remove', options, cb);
+            return cb(error || 'Invalid response');
+        }).form(options);
+    };
+
+
+    // generate all API methods
+    slackApiMethods.forEach(function(slackMethod) {
+        // most slack api methods are only two parts, of the form group.method, e.g. auth.test
```

Comment:
Sure thing, is there a jsdoc convention for inline docs like this? Most of what I know is for method signatures or class definitions.